sources/web/datalab/auth.ts

```diff
@@ -128,6 +128,17 @@ export function isSignedIn(): boolean {
   return (process.env.DATALAB_ENV != 'local' || fs.existsSync(appCredFile));
 }
 
+function setOauth2Client(parsed_url: any): void {
+  if (!oauth2Client) {
+    var OAuth2:any = google.auth.OAuth2;
+    // TODO(ojarjur): Ideally, we would get the host and port from the parsed_url
+    // rather than hard-coding them. However, the client ID and secret we are using
+    // are limited to localhost only. We should consider making this configurable, or
+    // using the OAuth flow for non-web applications.
+    oauth2Client = new OAuth2(clientId, clientSecret, 'http://localhost:8081/oauthcallback');
```

Comment:
That would be doable, but would prevent us from being able to use the parsed_url to build the callback in the future (as opposed to hardcoding 'localhost:8081').

I'm a little torn about whether or not to keep this flexibility that we aren't using yet, but I suspect we'll want to be able to change at least the port number (without recompiling) in the near future.